### PR TITLE
Use named volumes unless MZ_CHBENCH_SNAPSHOT is set

### PR DESCRIPTION
**Pull request by cirego**
We've been having issues with the `./tmp` directory causing startup
issues with Zookeeper, failures in CI and, by default, the directory is
created with root permissions.

When MZ_CHBENCH_SNAPSHOT is set, we can tack on the subdirectory using
":+" interpolation, preserving the existing behavior. And when
MZ_CHBENCH_SNAPSHOT is unset, we avoid the extra string, thereby allowing
us to use a named volume instead of `./tmp`.

This one has been bugging me for a while, so I figured it was time to go ahead
and fix it!

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/6290)

### CODE REVIEW

File: demo/chbench/mzcompose.yml [hunk 1]
```diff
@@ -343,9 +344,12 @@ mzworkflows:
...
   prepare-snapshot:
     env:
-      MZ_CHBENCH_SNAPSHOT: ${MZ_CHBENCH_SNAPSHOT:-./tmp}
-      MZ_CHBENCH_SNAPSHOT_KAFKA_DATA: ${MZ_CHBENCH_SNAPSHOT:-./tmp}/kafka-data
-      MZ_CHBENCH_SNAPSHOT_ZK_DATA: ${MZ_CHBENCH_SNAPSHOT:-./tmp}/zookeeper
+      # If MZ_CHBENCH_SNAPSHOT is defined, use a sub-directory within the snapshot directory
+      # Otherwise, use a named volume
+      # Docker compose doesn't know how to interpolate using ":+", so the interpolation must be
+      # done here, using mzcompose
+      MZ_CHBENCH_SNAPSHOT_KAFKA_DATA: ${MZ_CHBENCH_SNAPSHOT:-kafka-data}${MZ_CHBENCH_SNAPSHOT:+/kafka-data}
```
Comment (quodlibetor) [inline demo/chbench/mzcompose.yml:351]: should this not be `./kafka-data` (leading-dot) to make it a sub-directory?

File: demo/chbench/mzcompose.yml [hunk 2]
```diff
@@ -281,6 +281,7 @@ volumes:
 
   # shared volumes for load tests, but a mounted directories for benchmarks
```
Comment (quodlibetor) [inline demo/chbench/mzcompose.yml:282]: Now that there are multiple env vars that affect this, could you maybe document all of them? I'm not sure where the best place to do it is, but here seems fine.

File: demo/chbench/mzcompose.yml [hunk 3]
```diff
@@ -343,9 +344,12 @@ mzworkflows:
...
     env:
-      MZ_CHBENCH_SNAPSHOT: ${MZ_CHBENCH_SNAPSHOT:-./tmp}
-      MZ_CHBENCH_SNAPSHOT_KAFKA_DATA: ${MZ_CHBENCH_SNAPSHOT:-./tmp}/kafka-data
-      MZ_CHBENCH_SNAPSHOT_ZK_DATA: ${MZ_CHBENCH_SNAPSHOT:-./tmp}/zookeeper
+      # If MZ_CHBENCH_SNAPSHOT is defined, use a sub-directory within the snapshot directory
+      # Otherwise, use a named volume
+      # Docker compose doesn't know how to interpolate using ":+", so the interpolation must be
+      # done here, using mzcompose
+      MZ_CHBENCH_SNAPSHOT_KAFKA_DATA: ${MZ_CHBENCH_SNAPSHOT:-kafka-data}${MZ_CHBENCH_SNAPSHOT:+/kafka-data}
+      MZ_CHBENCH_SNAPSHOT_ZK_DATA: ${MZ_CHBENCH_SNAPSHOT:-zk-data}${MZ_CHBENCH_SNAPSHOT:+/zookeeper}
```
Comment (quodlibetor) [inline demo/chbench/mzcompose.yml:352]: same, I think that the second one should be `./zookeeper.`

File: demo/chbench/mzcompose.yml (hunk 1 above)
Comment (cirego) [reply demo/chbench/mzcompose.yml:351]: Which part? The first one should be the name of a volume and the second one should be an extension to the path, not a path relative to the current working directory.

File: demo/chbench/mzcompose.yml (hunk 2 above)
Comment (cirego) [reply demo/chbench/mzcompose.yml:282]: Sure, I can document this. The main environment variable is meant be MZ_CHBENCH_SNAPSHOT and the KAFKA/ZK ones are only meant to be internal to the mzcompose file.

File: demo/chbench/mzcompose.yml (hunk 3 above)
Comment (cirego) [reply demo/chbench/mzcompose.yml:352]: This one is meant to be a path extension, not a path relative to the current working directory.

**Pull request merged**